static/src/javascripts/projects/common/modules/identity/cookierefresh.ts

```diff
@@ -0,0 +1,47 @@
+import { storage } from '@guardian/libs';
+import {
+	getUserFromApiWithRefreshedCookie,
+	isUserLoggedIn,
+	refreshOktaSession,
+} from 'common/modules/identity/api';
+
+const days30InMillis: number = 1000 * 60 * 60 * 24 * 30;
+
+const shouldRefreshCookie: (
+	lastRefresh: string | null,
+	currentTime: number,
+) => boolean = (lastRefresh: string | null, currentTime: number) => {
+	// Cookies auto-refresh after 30 days
+	return (
+		!lastRefresh || currentTime - parseInt(lastRefresh, 10) > days30InMillis
+	);
+};
+
+const safelyGet = (value: unknown) => {
+	switch (typeof value) {
+		case 'string':
+			return value;
+		default:
+			return null;
+	}
+};
+
+const init: () => void = () => {
+	const lastRefreshKey = 'identity.lastRefresh';
+
+	if (storage.local.isAvailable() && isUserLoggedIn()) {
+		const currentTime: number = new Date().getTime();
+		const lastRefresh: string | null = safelyGet(
+			storage.local.get(lastRefreshKey),
+		);
+		if (shouldRefreshCookie(lastRefresh, currentTime)) {
+			getUserFromApiWithRefreshedCookie().catch(() => {
+				/* do nothing */
+			});
+			storage.local.set(lastRefreshKey, currentTime);
+			refreshOktaSession(encodeURIComponent(document.location.href));
+		}
+	}
+};
```

Comment:
The reason this broke and needed to be reverted in #25337 is because `storage.local.get` actually returns actual JS primitives, not simply a `string` as the native `storage.getItem` implementation.

I was the one who suggested converting this file to TypeScript, but using the `@guardian/libs`’s API actually introduced a new return value, which might have been difficult to identify. If I can offer some suggestion, I think that if we are to use the `@guardian/libs`, we should also use the built-in expiry mechanism:

```ts
const init: () => void = () => {
	const freshCookieKey = 'identity.freshCookie';

	if (storage.local.isAvailable() && isUserLoggedIn()) {
		const fresh: boolean = !!storage.local.get(freshCookieKey);
		if (!fresh) {
			getUserFromApiWithRefreshedCookie().catch(() => {
				/* do nothing */
			});
			const expires = new Date().getTime() + days30InMillis;
			storage.local.set(freshCookieKey, true, expires);
			refreshOktaSession(encodeURIComponent(document.location.href));
		}
	}
};
```

I’ve also raised a PR to use the `unknown` type for the `storage.local.get` response: https://github.com/guardian/libs/pull/383